### DISCUSSION TITLE
parameters for getMarkerFeatures

### DISCUSSION THREAD

**Discussion by yxiao832** [GreenleafLab/ArchR#868] [Questions / Documentation]
Hi, 

Could you please explain a bit more about  the difference of the usage for parameters "bias" and "normby" in the "getMarkerFeatures".  If I want to correct my gene marker features between samples using TSS score and fragments, should I put them in both parameters? 

Since your example only gives value to "bias" , I am a bit confused.

Thanks!
Yang

**Comment by rcorces** [comment]
These parameters are actually quite different despite having similar sounding names.
Marker features are identified by comparing your cells of interest to a bias-matched group of cells. The parameter passed to `bias` indicates how that matched group of cells should be selected.
`normBy` on the other hand performs a normalization on the matrix passed to `useMatrix` prior to performing the marker feature identification. This value depends on the type of matrix you are passing. For ex. you would not normalize a `MotifMatrix` by `ReadsInTSS` because it is not a counts-based matrix. This is why `normBy` defaults to `NULL`.

These parameters are quite distinct and should not default be set to the same values

**Reply by yxiao832** [reply]
Thank you for the answer! 

So if I want to use getMarkerFeatures to calculate gene score using GeneScoreMatrix (should be a counts-based matrix to my understanding), assigning "TSSEnrichment", "log10(nFrags)" to either "bias" or "normBy"  and leave the other one as "NULL", I should get the same result?

Thanks!
Yang

**Reply by rcorces** [reply]
No that is not the correct interpretation unless I'm misunderstanding you. If your matrix needs to be normalized, you need to use `normBy`. This is completely unrelated to finding a bias-matched set of cells. In the case of `GeneScoreMatrix` you do not need to normalize so you just provide arguments to `bias`. But you cannot use these parameters interchangeably - you would not normalize a matrix by the `TSSEnrichment`, at least not in any normal circumstance that I can think of.

**Reply by yxiao832** [reply]
I see. Got it now! Thanks!

**Reply by ZunpengLiu** [reply]
Thank you guys for having this discussion and detailed reply. What I am thinking is that, if we want to call differential peaks responsible for diseases, can we pass some covariates to the "bias", such as age, sex, et al, and normalise the peak matrix by "normalise=log10(nFrags)"- which should be a whole-genome normalisation based on counts (similar to Diffbind) at the same time?  Do you think that is reasonable?
